server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -655,4 +657,35 @@ public double parseDouble(String value, boolean roundUp, LongSupplier now) {
             return Double.parseDouble(value);
         }
     };
+
+    DocValueFormat TIME_SERIES_ID = new TimeSeriesIdDocValueFormat();
+
+    /**
+     * DocValues format for time series id.
+     */
+    class TimeSeriesIdDocValueFormat implements DocValueFormat {
+        private TimeSeriesIdDocValueFormat() {}
+
+        @Override
+        public String getWriteableName() {
+            return "tsid";
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) {}
+
+        @Override
+        public String toString() {
+            return "tsid";
+        }
+
+        @Override
+        public Object format(BytesRef value) {
+            try {
+                return TimeSeriesIdGenerator.parse(new BytesArray(value).streamInput());
```

Comment:
I chose to make `parse` static here so any code can decode a `_tsid` without knowing what mapping it came from. We could go the other way - it'd save a few bytes of encoding. But I don't think its worth it.